### DISCUSSION TITLE
Point of using $id identity schemas

### DISCUSSION THREAD

**Discussion by hyperknot** [ory/kratos#3028] [Q&A] [answered]
What is the point of using $id in identity schemas? 

I've seen the following:
- The minimal example doesn't have it:

```
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "type": "object",
  "properties": {
    "traits": {
      "type": "object"
    }
  }
}
```

- Most of the official examples all refer to an URL which doesn't exist in real life. For example https://schemas.ory.sh/presets/kratos/identity.basic.schema.json doesn't exist.

- In the session, I never see this $id anywhere.

So all in all, is there any point in having the $id filled in custom schemas?

**Comment by aeneasr** [comment, marked as answer]
You can leave it out if you want. It's only useful in some edge cases of JSON Schema itself :)
